### DISCUSSION TITLE
Support for ignoring differences in a field during reconcile

### DISCUSSION THREAD

**Discussion by dbbhat** [fluxcd/flux2#4225] [Q&A]
We are using Connaisseur for container image signature verification on our clusters, alongside with Flux2

There seems to be a limitation with Flux2 when working with Connaisseur (or any mutating webhook that modifies the image reference of a manifest) -

In our case, Connaisseur is deployed as a mutating admission controller and it mutates the image reference from `<image-name>:<tag>` in the Pod Spec (what's defined in git repo) to `<image-name>:<tag>@sha256<digest>` after successfully verifying image signature so that we can be sure that we are actually deploying the image whose signature has been verified by Connaisseur. See 
https://sse-secure-systems.github.io/connaisseur/v3.0.0/#trusted-digests for more details. 

However, during reconcile, Flux sees a drift in the `image:` field and updates it back to `<image>:<tag>` value defined in git repo. 

This results in an infinite cycle of `signature verification by Connaisseur` ->  `mutation of image reference by Connaisseur to include digest` -> `update of resource back to <image>:<tag> by Flux` because Flux assumes that cluster state is always out of sync with the remote  git repo.

Does Flux support a feature to ignore differences in a particular field (in our case `image:` field), something like `ignoreDifferences` which allows you to optionally ignore differences for updates for that field and avoid reconciling on seeing a drift in that field? The diffing customization can be configured for single or multiple application resources or at a system level.

Can support for something like this be added if it doesn't already exist?

**Comment by hcsaustrup** [comment]
I'm having the same issue, although with Vertical Pod Autoscaler - every time deployment resources are updated by VPA, flux comes along and rolls back the changes.
